src/ncp/Makefile.am

```diff
@@ -84,6 +85,16 @@ COMMON_SOURCES                                    = \
     ncp_uart.hpp                                    \
     $(NULL)
 
+if OPENTHREAD_ENABLE_NCP_VENDOR_HOOK
+COMMON_SOURCES                                   += \
+    ncp_vendor_hook_temp.cpp                        \
+    $(NULL)
+
+.INTERMEDIATE: ncp_vendor_hook_temp.cpp
+ncp_vendor_hook_temp.cpp: ${OPENTHREAD_NCP_VENDOR_HOOK_SOURCE}
```

Comment:
will ncp_vendor_hook_temp.cpp get deleted when compilation completes?

Reply:
Yep, that's what the `.INTERMEDIATE:` line does.